### ISSUE TITLE
use custom onCredAuth in Python

### ISSUE THREAD

**Issue by p1-afv** [pjsip/pjproject#4084]
### Describe the feature

I have python wrapper of pjsua2 but I need to use custom digest parameters calculattion function.
The doc says "if PJSIP_HAS_DIGEST_AKA_AUTH is disabled, you have to overload this callback to provide your own digest computation back-end.
If PJSIP_HAS_DIGEST_AKA_AUTH is enabled, libmilenage library from third_party directory is linked, and this callback returns PJ_ENOTSUP, then the default digest computation back-end is used."

Fine.. I have all build with PJSIP_HAS_DIGEST_AKA_AUTH  == 0 
and use dataType == PJSIP_CRED_DATA_EXT_AKA
Call back fucntion is overloaded.. but it never called as the code fail with
08:06:16.608            pjsua_acc.c  .Acc 0: setting registration..
python: ../src/pjsip/sip_auth_client.c:830: pjsip_auth_clt_set_credentials: Assertion `!"PJSIP_HAS_DIGEST_AKA_AUTH is not enabled"' failed.

I am trying to dig through C code and frankly dont understand assertion logic.

What I do wrong ?

cred = pj.AuthCredInfo("digest", "*", "xxxxxxx@ims.mnc000.mcc00.3gppnetwork.org", pj.PJSIP_CRED_DATA_EXT_AKA, "")

thanks in advance

btw, trying to build with  PJSIP_HAS_DIGEST_AKA_AUTH  == 1  never succeed

**Comment by nanangizz** [comment]
To enable digest AKA auth, `PJSIP_HAS_DIGEST_AKA_AUTH` must be set to 1 (please check [here](https://docs.pjsip.org/en/latest/api/generated/pjsip/group/group__PJSIP__AUTH__AKA__API.html) for more info). Just tried to fix the docs in #4089, thanks for pointing it out.

Now, could you describe more about the build failure when `PJSIP_HAS_DIGEST_AKA_AUTH == 1`.

**Comment by p1-afv** [comment]
> To enable digest AKA auth, `PJSIP_HAS_DIGEST_AKA_AUTH` must be set to 1 (please check [here](https://docs.pjsip.org/en/latest/api/generated/pjsip/group/group__PJSIP__AUTH__AKA__API.html) for more info). Just tried to fix the docs in #4089, thanks for pointing it out.
> 
> Now, could you describe more about the build failure when `PJSIP_HAS_DIGEST_AKA_AUTH == 1`.

Helo.. Yehh.. I have already guessed that PJSIP_HAS_DIGEST_AKA_AUTH is must to have to enable onCredAuth work.
I also solved an assue (for me) building project with the PJSIP_HAS_DIGEST_AKA_AUTH enabled.
I found that problem was that a) third_party/milenage doesnt compile (I meean make does nothing with it).. and even if I compile it manually it doesnt link afterword.. 
So I simply fixed couple of .mak files to make it working. Frankly I didint it in a very straightforward way, just adding milenage staticaly to be considered for compilation/linkage.

So, finaly I got onCredAuth callback function  involved in the authentication process. But.. with my I staticaly defined akaK, akaOp and akaAmf it causes error in milenage function.
As I mentioned I am using swig/python wrapper and something wrong there with data types. E.g. akaK supposed to be python str.. while in relaity it may include some non-pritable characters e.g 0x00 and python keep it  as "\x00" in the string and then it cause wrong akaK length in C function. I am not big expert in CPP/C and struggled to understand why it happen but I belibe in python these class properties  should be defined as bytes.

So I decided do not rely on built in milenage, but use CryptoMobile instead in my own onCredAuth()

p.s.. one more question so far.. I never used swig python wrappers.. and struggling to debug fuctions (in VSCode) which I overload..say onCredAuth. Breakpoint set wihin the function doesnt work.. Any suggestion how to make it working ?

thanks
